tests/devices/i10/test_i10Apple2.py

```diff
@@ -320,26 +349,99 @@ async def test_I10Apple2_pol_set(
         assert float(gap.call_args[0][0]) == pytest.approx(expect_gap, 0.05)
 
 
+async def test_linear_arbitrary_pol_fail(
+    mock_linear_arbitrary_angle: LinearArbitraryAngle,
+):
+    mock_linear_arbitrary_angle.id.pol = "lh"
+    with pytest.raises(RuntimeError) as e:
+        await mock_linear_arbitrary_angle.set(20)
+    assert str(e.value) == (
+        f"Angle control is not available in polarisation"
+        f" {mock_linear_arbitrary_angle.id.pol} with {mock_linear_arbitrary_angle.id.name}"
+    )
+
+
+@pytest.mark.parametrize(
+    "poly",
+    [18, -18, 12.01, -12.01],
+)
+async def test_linear_arbitrary_limit_fail(
+    mock_linear_arbitrary_angle: LinearArbitraryAngle, poly: float
+):
+    mock_linear_arbitrary_angle.id.pol = "la"
+    mock_linear_arbitrary_angle.jawphase_from_angle = poly1d([poly])
+    with pytest.raises(RuntimeError) as e:
+        await mock_linear_arbitrary_angle.set(20)
+    assert (
+        str(e.value) == f"jawphase position for angle (20) is outside permitted range"
+        f" [-{mock_linear_arbitrary_angle.jawphase_limit}, {mock_linear_arbitrary_angle.jawphase_limit}]"
+    )
+
+
+async def test_linear_arbitrary_RE_scan(
+    mock_linear_arbitrary_angle: LinearArbitraryAngle, RE: RunEngine
+):
+    docs = defaultdict(list)
+    rbv_mocks = Mock()
+    rbv_mocks.get.side_effect = range(0, 181, 10)
+    callback_on_mock_put(
+        mock_linear_arbitrary_angle.id.id_jaw_phase.jaw_Phase.user_setpoint,
+        lambda *_, **__: set_mock_value(
+            mock_linear_arbitrary_angle.id.id_jaw_phase.jaw_Phase.user_setpoint_readback,
+            rbv_mocks.get(),
+        ),
+    )
+
+    def capture_emitted(name, doc):
+        docs[name].append(doc)
+
+    mock_linear_arbitrary_angle.id.pol = "la"
+    RE(
+        scan(
+            [mock_linear_arbitrary_angle], mock_linear_arbitrary_angle, 0, 180, num=11
+        ),
+        capture_emitted,
+    )
+    assert_emitted(docs, start=1, descriptor=1, event=11, stop=1)
+
+    jawphase = get_mock_put(
+        mock_linear_arbitrary_angle.id.id_jaw_phase.jaw_Phase.user_setpoint
+    )
+
+    poly = poly1d([1.0 / 7.5, -120.0 / 7.5])
+    for cnt, data in enumerate(docs["event"]):
+        temp_angle = 18 * cnt
+        assert data["data"]["mock_linear_arbitrary_angle-angle"] == temp_angle
+        alpha_real = (
+            temp_angle
+            if temp_angle > mock_linear_arbitrary_angle.angle_threshold_deg
+            else temp_angle + 180.0
+        )
+        assert jawphase.call_args_list[cnt] == mock.call(
+            str(poly(alpha_real)), wait=True, timeout=mock.ANY
+        )
+
+
 @pytest.mark.parametrize(
     "fileName, expected_dict, source",
     [
         (
-            IDGAPLOOKUPTABLE,
+            ID_GAP_LOOKUP_TABLE,
             "tests/devices/i10/lookupTables/expectedIDEnergy2GapCalibrationsIdu.pkl",
```

Comment:
should: extract the folder part of the path so that we can clearly see that only the filename is different